transactions.go

```diff
@@ -138,10 +153,8 @@ func (t *Transactions) ReshufflePending(next Block) []TransactionID {
 
 	// Delete mempool entries for transactions in the finalized block.
 
-	lookup := make(map[TransactionID]struct{})
-
 	for _, id := range next.Transactions {
-		lookup[id] = struct{}{}
+		t.finalized[id] = struct{}{}
```

Comment:
Rather than keep a separate `finalized` index, why not just check `t.buffer`? If I recall, `t.buffer` is persisted and loaded on startup with all finalized transactions from the database in `(*Ledger).loadTransactions()`.

Reply:
`loadTransactions` loads transactions to buffer only in case they are stored in db. if node is restarted and then synced, those transactions won't be in db, but they will be synced. so we can mark them as finalized - since we only have id at this point - and then safely sync them along any other new transaction.
also in buffer we store all transactions, both finalized and not, so we need somehow to distinguish them.

Reply:
Hm, okay :+1:.